packages/web/server/api/pagination.ts

```diff
@@ -35,3 +40,38 @@ export function maybeCursorPaginatedItems<TItem>(
     nextCursor: cursor + limit > items.length - 1 ? null : cursor + limit,
   };
 }
+
+const paginationCache = new LRUCache<string, CacheEntry>(DEFAULT_LRU_OPTIONS);
+/** Returns cached items while the user is paginating the list.
+ *  If the input changes or the cursor is reset, the cache is invalidated.
+ *  With useInfiniteQuery, the cursor will be reset to 0 when input to the query changes or the query is invalidated. */
+export async function maybeCachePaginatedItems<TItem>(
+  getFreshItems: () => Promise<TItem[]>,
+  cacheKey: string,
+  cursor: number | null | undefined,
+  limit: number | null | undefined
+): Promise<{
```

Comment:
It's a little hard to keep track of the parameters. What are your thoughts on making this an parameter object?